neuralop/training/callbacks.py

```diff
@@ -375,82 +376,123 @@ def on_before_loss(self, out):
     
     def on_before_val_loss(self, **kwargs):
         return self.on_before_loss(**kwargs)
-
-class ModelCheckpointCallback(Callback):
-
-    def __init__(self, checkpoint_dir: Union[Path, str] = Path('./checkpoints'), interval: int = 1):
-        """
-        Implements basic model checkpointing by saving a model every N epochs.
         
+class StateCheckpointCallback(Callback):
+    
+    def __init__(self, 
+                 save_dir: Union[Path, str] = './checkpoints', 
+                 save_best : str = None,
+                 save_step : int = 1,
+                 save_optimizer : bool = False,
+                 save_scheduler : bool = False,
+                 save_regularizer : bool = False,
+                 resume_from_dir : Union[Path, str] = None
+                 ):
+        """StateCheckpointCallback handles saving and resuming 
+        training state from checkpoint .pt save files. 
+
         Parameters
         ----------
-        checkpoint_dir : str | pathlib.Path
-            folder in which to save checkpoints
-        interval : int
-            interval at which to check metric
+        save_dir : Union[Path, str], optional
```

Comment:
No need to use format syntax in docstring, no need for Union, Path or str is fine